linux/devdoc/file_linux_requirements.md

```diff
@@ -0,0 +1,86 @@
+# file_linux
+================
+
+## Overview
+
+Linux implementation of the `file` module.
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, FILE_HANDLE, file_create, EXECUTION_ENGINE_HANDLE, execution_engine, const char*, full_file_name, uint64_t, desired_file_size, bool, has_manage_volume);
+MOCKABLE_FUNCTION(, void, file_destroy, FILE_HANDLE, handle);
+MOCKABLE_FUNCTION_WITH_RETURNS(, FILE_WRITE_ASYNC_RESULT, file_write_async, FILE_HANDLE, handle, CONSTBUFFER_HANDLE, source, uint64_t, position, FILE_WRITE_CB, user_callback, void*, user_context)(0, MU_FAILURE);
+MOCKABLE_FUNCTION_WITH_RETURNS(, FILE_READ_ASYNC_RESULT, file_read_async, FILE_HANDLE, handle, uint32_t, size, uint64_t, position, FILE_READ_CB, user_callback, void*, user_context)(0, MU_FAILURE);
+MOCKABLE_FUNCTION_WITH_RETURNS(, int, file_extend_filesize, FILE_HANDLE, handle, uint64_t, desired_size, bool, has_manage_volume)(0, MU_FAILURE);
+```
+
+## file_create
+
+```c
+MOCKABLE_FUNCTION(, FILE_HANDLE, file_create, EXECUTION_ENGINE_HANDLE, execution_engine, const char*, full_file_name, uint64_t, desired_file_size, bool, has_manage_volume);
+```
+
+**SRS_FILE_LINUX_43_001: [** `file_create` shall call `open` with `full_file_name` as `pathname` and flags `O_CREAT` and `O_RDWR`. **]**
+
+**SRS_FILE_LINUX_43_002: [** `file_create` shall return the file handle returned by the call to open. [`open` documentation](https://www.man7.org/linux/man-pages/man2/open.2.html) **]**
+
+## file_destroy
+
+```c
+MOCKABLE_FUNCTION(, void, file_destroy, FILE_HANDLE, handle);
+```
+
+**SRS_FILE_LINUX_43_003: [** `file_destroy` shall call `close` with `fd` as `handle`. [`close` documentation](https://www.man7.org/linux/man-pages/man2/close.2.html) **]**
+
+## file_write_async
+
+```c
+MOCKABLE_FUNCTION_WITH_RETURNS(, FILE_WRITE_ASYNC_RESULT, file_write_async, FILE_HANDLE, handle, CONSTBUFFER_HANDLE, source, uint64_t, position, FILE_WRITE_CB, user_callback, void*, user_context)(0, MU_FAILURE);
+```
+
+**SRS_FILE_LINUX_43_016: [** `file_write_async` shall create a `FILE_LINUX_IO` struct with `handle` as `handle`, `FILE_ASYNC_WRITE` as `type`. **]**
+
+**SRS_FILE_LINUX_43_017: [** `file_write_async` shall populate the `data` field of the `FILE_LINUX_IO` struct using `source`, `user_callback` and `user_context`. **]**
+
+**SRS_FILE_LINUX_43_004: [** `file_write_async` shall initialize a `sigevent` struct with `SIGEV_THREAD` as `sigev_notify`, the created `FILE_LINUX_IO` struct as `sigev_value`, `on_file_io_complete_linux` as `sigev_notify_function`, `NULL` as `sigev_notify_attributes`(unsure about what `sigev_notify_attributes` should be). [`sigevent` documentation](https://man7.org/linux/man-pages/man7/sigevent.7.html). **]**

```

Comment:
>on_file_io_complete_linux [](start = 178, length = 25)

Missing specs for this function.
Also, we can use separate on_read_complete and on_write_complete functions with the Linux APIs (we only have a shared one in Windows because that's how the win32 API's work).